parquet/src/file/serialized_reader.rs

```diff
@@ -704,105 +703,76 @@ impl SerializedPageReaderContext {
             inner: input,
             bytes_read: 0,
         };
-        let header = self.read_page_header(&mut tracked, page_index, dictionary_page)?;
+        let header = context.read_page_header(&mut tracked, page_index, dictionary_page)?;
         Ok((tracked.bytes_read, header))
     }
 
+    fn read_page_header_len_from_bytes(
+        context: &SerializedPageReaderContext,
+        buffer: &[u8],
+        page_index: usize,
+        dictionary_page: bool,
+    ) -> Result<(usize, PageHeader)> {
+        let mut input = std::io::Cursor::new(buffer);
```

Comment:
:+1: nice, yeah it makes sense to just use the same approach with and without encryption to get the header + length here rather than use `TCompactSliceInputProtocol`.